hermes/bin/src/runtime_extensions/hermes/cron/state.rs

```diff
@@ -0,0 +1,381 @@
+//! Cron State.
+/// The crontabs hash map.
+use std::hash::{Hash, Hasher};
+
+use once_cell::sync::Lazy;
+use tokio::{
+    runtime::Builder,
+    sync::{mpsc, oneshot},
+};
+
+use super::{
+    event::OnCronEvent,
+    queue::{cron_queue_task, CronEventQueue, CronJob, CronJobDelay},
+};
+use crate::runtime_extensions::{
+    bindings::hermes::cron::api::{CronEventTag, CronTagged, Instant},
+    hermes::cron::mkdelay_crontab,
+};
+
+/// Cron Internal State
+static CRON_INTERNAL_STATE: Lazy<InternalState> = Lazy::new(|| {
+    let sender = if let Ok(runtime) = Builder::new_current_thread().enable_all().build() {
+        let (sender, receiver) = mpsc::channel(1);
+
+        let _handle = std::thread::spawn(move || {
+            runtime.block_on(async move {
+                let h = tokio::spawn(cron_queue_task(receiver));
+                drop(tokio::join!(h));
+            });
+            std::process::exit(0);
+        });
+        Some(sender)
+    } else {
+        // TODO (@saibatizoku): log error https://github.com/input-output-hk/hermes/issues/15
+        // Failed to start the queue task
+        None
+    };
+
+    InternalState::new(sender)
+});
+
+/// Name of the Application that owns the `OnCronEvent`s.
+pub type AppName = String;
+
+/// Internal State.
+pub(crate) struct InternalState {
+    /// The send events to the crontab queue.
+    pub(crate) cron_queue: CronEventQueue,
```

Comment:
```suggestion
    cron_queue: CronEventQueue,
```

We shouldn't export the `cron_queue` directly, and it doesn't;t appear that its used anyway,

Reply:
Yes, thanks for the catch.